File: src/data_structure/linked_list/delete.cpp
/*
  Delete Node at a given position in a linked list 
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/
Node* Delete(Node *head, int position)
{
    if (head == NULL) return NULL;
    if (position == 0)
    {
        Node *c = head->next;
        free(head);
        return c;
    }
    Node *pre = head, *cur = head;
    int i = 0;
    while (cur != NULL && i++ < position)
    {
        pre = cur;
        cur = cur->next;
    }
    pre->next = cur->next;
    free(cur);
    return head;
}

File: src/data_structure/linked_list/delete.java
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
    // This is a "method-only" submission. 
    // You only need to complete this method. 

Node Delete(Node head, int position) {
    if (head == null) {
        return null;
    }
    if (position == 0) {
        Node n = head.next;
        head = null;
        return n;
    }
    Node cur = head, prev = head;
    int i = 0;
    while (cur != null && i++ < position) {
        prev = cur;
        cur = cur.next;
    }
    prev.next = cur.next;
    cur = null;
    return head;
}


